src/engine/behavior_script.c

```diff
@@ -820,6 +820,13 @@ void cur_obj_update(void) {
     BhvCommandProc bhvCmdProc;
     s32 bhvProcResult;
 
+    if (!(objFlags & OBJ_FLAG_PROCESS_OUTSIDE_ROOM)) {
+        if (o->oRoom != -1 && gMarioCurrentRoom != 0 && !is_room_loaded(o->oRoom)) {
+            cur_obj_enable_rendering_if_mario_in_room();
```

Comment:
Slight nitpick, but it probably makes more sense to copy the three lines out of the `else` block in this function to skip the three unnecessary conditions.